### PR TITLE
Replace out of date publish images with cgr.dev equivalents.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

These should be drop in replacements with full shells.

Grype Output:

gcr.io/go-containerregistry/crane:debug | cgr.dev/chainguard/crane:latest-dev
-- | --
1 Critical<br>5 High<br>10 Medium<br>11 Negligible<br>2 Unknown | 1 Medium

golang:1.18.7 | cgr.dev/chainguard/go:latest-dev
-- | --
140 Critical<br>416 High<br>19 Low<br>337 Medium<br>452 Negligible<br>8 Unknown | 

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by imjasonh** [comment]
FWIW, it looks like `gcr.io/go-containerregistry/crane:debug` hasn't been updated since Mon Jul 31 2023 21:49:55, which is likely when the repo stopped having access to GCP. I don't know the current ownership of that project.

If there's interest, I think we could publish crane images to `ghcr.io/google/go-containerregistry:debug` where it can continue getting updates. Until then, the cgr.dev image is probably going to be the best place to get updates for now.

File: tekton/publish.yaml [hunk 1]
```diff
@@ -80,7 +80,7 @@ spec:
       cp ${DOCKER_CONFIG} /workspace/docker-config.json
 
   - name: create-ko-yaml
-    image: golang:1.18.7
+    image: cgr.dev/chainguard/go:latest-dev
```
Comment (afrittoli) [inline tekton/publish.yaml:83]: I think we need a version here instead of `latest-dev`

File: tekton/publish.yaml [hunk 2]
```diff
@@ -61,7 +61,7 @@ spec:
   steps:
 
   - name: container-registy-auth
-    image: gcr.io/go-containerregistry/crane:debug
+    image: cgr.dev/chainguard/crane:latest-dev
```
Comment (afrittoli) [inline tekton/publish.yaml:64]: We don't pin crane today, which is probably ok, even though this means that new crane releases might break the release process

File: tekton/publish.yaml (hunk 1 above)
Comment (wlynch) [reply tekton/publish.yaml]: Versioned tags aren't part of free tier. Pinned the digest + enabled dependabot!

File: tekton/publish.yaml (hunk 2 above)
Comment (wlynch) [reply tekton/publish.yaml]: Done.

**Review by vdemeester — APPROVED** [review]

**Comment by Yongxuanzhang** [comment]
Cool! So the images are created and hosted by chainguard? How do they fix those vulberabilities?

**Comment by wlynch** [comment]
> Cool! So the images are created and hosted by chainguard?

Yep! https://edu.chainguard.dev/chainguard/chainguard-images/overview/ for more details.

>  How do they fix those vulberabilities?

Magic 🧙 
(really a lot of hard work and automation 💪 🤖 )

**Review by Yongxuanzhang — APPROVED** [review]

**Comment by vdemeester** [comment]
@wlynch @imjasonh still needed ? (if yes, it needs a rebase 🙏🏼 )

**Comment by afrittoli** [comment]
@wlynch ping 🙏

**Comment by wlynch** [comment]
rebased!

**Review by vdemeester — APPROVED** [review]
/cc @tektoncd/core-maintainers

**Review by afrittoli — COMMENTED** [review]
/lgtm

**Pull request merged**